afm/server.py

```diff
@@ -56,13 +57,21 @@ def _filter_columns(self, schema, columns):
         return pa.schema([pa.field(f.name, f.type, f.nullable, f.metadata) for f in fields])
 
     # write arrow dataset to filesystem
-    def _write_asset(self, asset, reader):
+    def _write_asset(self, asset, reader, write_mode):
         # in this implementation we currently begin by reading the entire dataset
         record_batches = reader.read_all().combine_chunks().to_batches()
         transformed_batches = transform_batches(asset.actions, record_batches)
-        # currently, write_dataset supports the parquet format, but not csv
-        ds.write_dataset(transformed_batches, base_dir=asset.path, format=asset.format,
-                         filesystem=asset.filesystem)
+        # If the client's request is to append to the existing data then the flag `overwrite_or_ignore`
+        # is used with a unique basename_template that is related to the time of writing.
+        # This operation writes the data to a new file while ignoring (not changing) the existed files.
+        # Otherwise, the flag `delete_mathing` is used. This operation writes the data to a new file
+        # while deleting the existed files.
+        if write_mode == "append":
+            logger.trace("write_mode: append", extra={DataSetID: asset.name, ForUser: True})
+            ds.write_dataset(transformed_batches, base_dir=asset.path, basename_template="part-{:%Y-%m-%d-%H-%M-%S-%f}-{{i}}.parquet".format(datetime.datetime.now()), format=asset.format,  filesystem=asset.filesystem, existing_data_behavior='overwrite_or_ignore')
+        else:
+            logger.trace("write_mode: overwrite", extra={DataSetID: asset.name, ForUser: True})
+            ds.write_dataset(transformed_batches, base_dir=asset.path, basename_template="part-{:%Y-%m-%d-%H-%M-%S-%f}-{{i}}.parquet".format(datetime.datetime.now()), format=asset.format, filesystem=asset.filesystem, existing_data_behavior='delete_matching')
```

Comment:
lines 71-71 are almost identical to 73-74. Try to merge them

Reply:
Fixed thanks